platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/merkle/VirtualRootNode.java

```diff
@@ -1198,10 +1199,10 @@ public boolean isSelfHashing() {
      */
     @Override
     public Hash getHash() {
-        if (super.getHash() == null) {
+        if (hash.get() == null) {
             pipeline.hashCopy(this);
         }
-        return super.getHash();
+        return hash.get();
```

Comment:
Just double checking. Is `pipeline.hashCopy()` thread safe if called simultaneously by different threads?

Reply:
Yes, it is. We had issues in the past caused by calling `pipeline.hashCopy()` in parallel from many threads, this is definitely a possible scenario.